mycroft/skills/skill_loader.py

```diff
@@ -41,15 +41,15 @@ def remove_submodule_refs(module_name):
         module_name: name of skill module.
     """
     submodules = []
-    LOG.debug('Skill module: {}'.format(module_name))
+    LOG.debug(f'Skill module: {module_name}')
     # Collect found submodules
     for m in sys.modules:
         if m.startswith(module_name + '.'):
             submodules.append(m)
     # Remove all references them to in sys.modules
     for m in submodules:
-        LOG.debug('Removing sys.modules ref for {}'.format(m))
-        del(sys.modules[m])
+        LOG.debug(f'Removing sys.modules ref for {m}')
+        del (sys.modules[m])
```

Comment:
nitpick, I think we can remove the parenthesis completely since del isn't a function but a keyword

Reply:
> I sort of miss the possibility to specify if a skill class will work for this version of the software to give the possibility to handle backwards compatibility. 

Can you elaborate on this? I see this PR as an optimization not a new feature, the aim is to be fully backwards compatible all the time. If all skills load fine what is there to handle?

A change like that would need to go into mycroft-core imho (your original PR!), in ovos-core we are limited in the scope of changes, in particular we can't expose any new user facing api in the skill class, otherwise a skill using that will not work in mycroft-core (for this use case we have ovos_workshop)